source/upgrade.txt

```diff
@@ -0,0 +1,115 @@
+  :ref:`<csharp-server-release-changes>` section.
+
+.. tip::
+
+   To minimize the amount of changes your application may require when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <csharp-stable-api>`.
+
+.. _csharp-breaking-changes:
+
+Breaking Changes
+----------------
+
+A breaking change is a modification in a convention or behavior in
+a specific version of the driver that may prevent your application from
+working properly if not addressed before upgrading.
+
+The breaking changes in this section are categorized by the driver version that
+introduced them. When upgrading driver versions, address all the breaking
+changes between the current and upgrade versions. For example, if you
+are upgrading the driver from v2.0 to v2.20, address all breaking changes from
+the version after v2.0 including any listed under v2.20.
+
+Version 2.19.0 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The ``ObjectSerializer`` has been changed to only allow deserialization of
+  types that are considered safe. What types are considered safe is
+  determined by a new configurable ``AllowedTypes`` function (of type
+  ``Func<Type, bool>``). The default ``AllowedTypes`` function is
+  ``ObjectSerializer.DefaultAllowedTypes`` which returns true for a
+  number of well-known framework types that we have deemed safe.
+  A typical example might be to allow all the default allowed types as
+  well as your own types. This could be accomplished as follows:
+
+  .. code-block:: csharp
+
+     var objectSerializer = new ObjectSerializer(
+        type => ObjectSerializer.DefaultAllowedTypes(type) || type.FullName.StartsWith("MyNamespace"));
+     BsonSerializer.RegisterSerializer(objectSerializer);
+    
+  More information about the ``ObjectSerializer`` can be found in our
+  :ref:`FAQ <csharp-faq-object-serializer>`.
+
+- The default ``LinqProvider`` has been changed to LINQ3. The ``LinqProvider``
+  can be changed back to LINQ2 in the following way:
+  
+  .. code-block:: csharp
+    
+     var connectionString = "<YourMongoDBConnectionString>";
+     var clientSettings = MongoClientSettings
+        .FromConnectionString(connectionString);
+     clientSettings.LinqProvider = LinqProvider.V2;
+     var client = new MongoClient(clientSettings);
+
+
+.. _csharp-server-release-changes:
+
+Server Release Compatibility Changes
+------------------------------------
+
+A server release compatibility change is a modification
+to the MongoDB .NET/C# Driver that discontinues support for a set of
+MongoDB Server versions.
+
+The driver discontinues support for a MongoDB Server version after it
+reaches end-of-life(EOL).
+
+To learn more about the MongoDB support for EOL products see the `Legacy
+Support Policy <https://www.mongodb.com/support-policy/legacy>`__. 
+
+To learn more about the compatibility between .NET/C# driver versions
+and MongoDB Server versions, visit the
+:ref:`Compatibility page<csharp-compatibility-tables>`. 
```

Comment:
Suggestion:

While this renders correctly, I think it's better to add a space between the text and directive name. Also, generally the link text should contain the title rather than "page", similar to the prior link to the compatibility tables.

```suggestion
:ref:`Compatibility <csharp-compatibility-tables>` page. 
```